src/character.cpp

```diff
@@ -2757,11 +2757,13 @@ float Character::fine_detail_vision_mod( const tripoint &p ) const
     float ambient_light{};
     tripoint const check_p = p == tripoint_min ? pos() : p;
     tripoint const avatar_p = get_avatar().pos();
-    // Light might not have been calculated on the NPC's z-level
+    map &here = get_map();
     if( is_avatar() || check_p.z == avatar_p.z ||
         ( fov_3d && std::abs( avatar_p.z - check_p.z ) <= fov_3d_z_range ) ) {
-        ambient_light = std::max( 1.0f,
-                                  LIGHT_AMBIENT_LIT - get_map().ambient_light_at( check_p ) + 1.0f );
+        if( !is_avatar() && here.ambient_light_at( check_p ) <= 4 ) {
```

Comment:
What's this 4? Did you mean `LIGHT_AMBIENT_DIM` (=5)? It shouldn't be needed if you can prevent recaching though.